### PR TITLE
[FLINK-30999] Introduce flink-table-store-test-utils for table store

### PR DESCRIPTION
**Pull request by FangYongs**
Introduce flink-table-store-test-utils module for table store

### CODE REVIEW

File: flink-table-store-test-utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>flink-table-store-parent</artifactId>
+        <groupId>org.apache.flink</groupId>
+        <version>0.4-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>flink-table-store-test-utils</artifactId>
+    <name>Flink Table Store: Test utils : </name>
```
Comment (SteNicholas) [inline flink-table-store-test-utils/pom.xml:32]:
```suggestion
    <name>Flink Table Store : Test Utils</name>
```

File: flink-table-store-test-utils/src/main/java/org/apache/flink/table/store/testutils/junit/TestLoggerExtension.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                context.getRequiredTestClass().getCanonicalName(),
+                context.getRequiredTestMethod().getName(),
+                context.getDisplayName());
+    }
+
+    @Override
+    public void testFailed(ExtensionContext context, Throwable cause) {
+        LOG.error(
+                "\n--------------------------------------------------------------------------------"
+                        + "\nTest {}.{}[{}] failed with:\n{}"
```
Comment (SteNicholas) [inline flink-table-store-test-utils/src/main/java/org/apache/flink/table/store/testutils/junit/TestLoggerExtension.java:60]:
```suggestion
                        + "\nTest {}.{}[{}] failed with:\n{}."
```

File: flink-table-store-test-utils/src/main/java/org/apache/flink/table/store/testutils/junit/parameterized/Parameters.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * The annotation is used to replace {@code Parameterized.Parameters} (Junit4) for Junit 5
+ * parameterized tests.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Parameters {
+    String name() default "{index}";
```
Comment (SteNicholas) [inline flink-table-store-test-utils/src/main/java/org/apache/flink/table/store/testutils/junit/parameterized/Parameters.java:33]: Adds the java doc of this method:
```
/**
   * Optional pattern to derive the test's name from the parameters. Use
   * numbers in braces to refer to the parameters or the additional data
   * as follows:
   * <pre>
   * {index} - the current parameter index
   * {0} - the first parameter value
   * {1} - the second parameter value
   * etc...
   * </pre>
   * <p>
   * Default value is "{index}" for compatibility with previous JUnit
   * versions.
   *
   * @return {@link MessageFormat} pattern string, except the index
   *         placeholder.
   * @see MessageFormat
   */
```

File: flink-table-store-test-utils/src/main/java/org/apache/flink/table/store/testutils/junit/parameterized/Parameter.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * The annotation is used to replace {@code Parameterized.Parameter} for Junit 5 parameterized
+ * tests.
+ */
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Parameter {
+    int value() default 0;
```
Comment (SteNicholas) [inline flink-table-store-test-utils/src/main/java/org/apache/flink/table/store/testutils/junit/parameterized/Parameter.java:33]: Adds the java doc of this method:
```
/**
   * Method that returns the index of the parameter in the array
   * returned by the method annotated by <code>Parameters</code>.
   * Index range must start at 0.
   * Default value is 0.
   *
   * @return the index of the parameter.
   */
```

**Comment by FangYongs** [comment]
Thanks @SteNicholas  DONE

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Comment by JingsongLi** [comment]
Where to use `test-utils`? Maybe we can modify the usage too?

**Comment by FangYongs** [comment]
> 

@JingsongLi It's used to upgrade `junit` and other common tests, I have update tests in `flink-table-store-common`

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**